### ISSUE TITLE
Telescope cannot create the history file

### ISSUE THREAD

**Issue by xarthurx** [nvim-telescope/telescope.nvim#2651]
### Description

Also reported here:  https://www.reddit.com/r/neovim/comments/v6towp/telescope_tries_to_create_cc_folder/

When doing a clean install, and enable the history functionality, telescope is unable to create the new folder/file.

![image](https://github.com/nvim-telescope/telescope.nvim/assets/1921878/2a843800-488f-4756-a91c-575bc9b6d63d)

The direct behavior is after entering the TS UI, I cannot get out. (by either pressing <ESC> or <Enter> to select some files).

### Neovim version

```markdown
NVIM v0.9.1
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

### Operating system and version

Windows

### Telescope version / branch / rev

2d92125620417fbea82ec30303823e3cd69e90e8

### checkhealth telescope

```markdown
==============================================================================
telescope: 

- ERROR No healthcheck found for "telescope" plugin.
```

### Steps to reproduce

1. clean install nvim
2. install `telescope`
3. run telescope find-file, find-history, etc.

### Actual behavior

no errors, the history file/folder should be created correctly.

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          { 'nvim-telescope/telescope-fzf-native.nvim', run = 'make' },
        },
      },
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup()
  require('telescope').load_extension('fzf')
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by xarthurx** [comment]
manually creating `telescope-history` will also not work.
As somehow telescope tries to concatenate the current dir with the `telescope-history` dir.
Is this a `plenary` problem?

**Comment by xarthurx** [comment]
OK, I found the problem:

I use `set shellslash` on Windows, to avoid the issue that I cannot open file paths with `()`.
This has no problem if TS has been run, but will create the above PATH issue for clean system.

What I did finally is:
1. unset `shellslash`
2. run TS once
3. set `shellslash` back.

Now everything works.

**Comment by DrKJeff16** [comment]
> relative issue about the `shellslash` topic: [neovim/neovim#24542](https://github.com/neovim/neovim/issues/24542)

In a config file or every time you open Nvim? Could you give an example? I'm trying to do the same but It doesn't result for me.

**Comment by xarthurx** [comment]
> 

You can set it in your config file, or `:set shellslash`.
You need to delete the whole `nvim-data` folder if you're on WIndows.

**Comment by Conni2461** [comment]
yeah this is a plenary.path problem: https://github.com/nvim-lua/plenary.nvim/issues/254

it was built without shellslash in mind. You can maybe try https://github.com/nvim-lua/plenary.nvim/pull/371 (note you probably want to rebase the PR its kinda old), so we can maybe find a working solution for this problem.

I just dont have a windows machine at hand to do the required testing

**Comment by xarthurx** [comment]
> yeah this is a plenary.path problem: [nvim-lua/plenary.nvim#254](https://github.com/nvim-lua/plenary.nvim/issues/254)
> 
> it was built without shellslash in mind. You can maybe try [nvim-lua/plenary.nvim#371](https://github.com/nvim-lua/plenary.nvim/pull/371) (note you probably want to rebase the PR its kinda old), so we can maybe find a working solution for this problem.
> 
> I just dont have a windows machine at hand to do the required testing

I wrote on that PR to get some attentions.

For Windows machine, perhaps a Virtual Machine?
